src/util/container.h

```diff
@@ -376,21 +376,33 @@ class ModifySafeMap
 			assert(false);
 			return;
 		}
-		if (m_iterating)
-			m_new.emplace(key, value);
-		else
-			m_values.emplace(key, value);
+		if (m_iterating) {
+			auto it = m_values.find(key);
+			if (it != m_values.end()) {
+				it->second = V();
```

Comment:
Sorry I am missing something. I'm sure there's a good reason, but why do we need to set ->second to a new value when we will set right below?

Reply:
This modifies `m_values`, below is `m_new`